### ISSUE TITLE
Getting ParserError when running Fetcher on a list of stocks

### ISSUE THREAD

**Issue by r8drascal** [AndrewRPorter/yahoo-historical#22]
When I run the Fetcher with a list of stocks, I keep getting this ParserError but it's not consistent. It just happens randomly. I tried putting an exemption and attempt the same stock again in a loop and it would sometimes work after a few tries, but later it would indefinitely get stuck on one of the stocks.

```
ParserError                               Traceback (most recent call last)
<ipython-input-15-6c36ca50f58c> in <module>
      2 
      3 print('Current time before download:', datetime.datetime.now())
----> 4 get_data(all_stocks,unqualified_stock_list,stock_list_idx=0)
      5 print('Current time after download:', datetime.datetime.now())

<ipython-input-13-f19ec7bb5b8b> in get_data(stock_list, unqualified_stock_list, save_limit, stock_list_idx)
      9         stock = stock_list[stock_list_idx]
     10         try:
---> 11             stock_data = Fetcher(stock, [stock_start_y,stock_start_m,stock_start_d],[stock_end_y,stock_end_m,stock_end_d]).getHistorical().values
     12         except KeyError:
     13             print('KeyError... Restarting Loop...')

~\Anaconda3\envs\tfgpu\lib\site-packages\yahoo_historical\fetch.py in getHistorical(self)
     49             DeprecationWarning,
     50         )
---> 51         return self._get("history")
     52 
     53     def getDividends(self):

~\Anaconda3\envs\tfgpu\lib\site-packages\yahoo_historical\fetch.py in _get(self, events)
     34         data = requests.get(url, headers={"User-agent": ""})
     35         content = StringIO(data.content.decode("utf-8"))
---> 36         return pd.read_csv(content, sep=",")
     37 
     38     def getData(self, events):

~\Anaconda3\envs\tfgpu\lib\site-packages\pandas\io\parsers.py in parser_f(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, dialect, error_bad_lines, warn_bad_lines, delim_whitespace, low_memory, memory_map, float_precision)
    674         )
    675 
--> 676         return _read(filepath_or_buffer, kwds)
    677 
    678     parser_f.__name__ = name

~\Anaconda3\envs\tfgpu\lib\site-packages\pandas\io\parsers.py in _read(filepath_or_buffer, kwds)
    452 
    453     try:
--> 454         data = parser.read(nrows)
    455     finally:
    456         parser.close()

~\Anaconda3\envs\tfgpu\lib\site-packages\pandas\io\parsers.py in read(self, nrows)
   1131     def read(self, nrows=None):
   1132         nrows = _validate_integer("nrows", nrows)
-> 1133         ret = self._engine.read(nrows)
   1134 
   1135         # May alter columns / col_dict

~\Anaconda3\envs\tfgpu\lib\site-packages\pandas\io\parsers.py in read(self, nrows)
   2035     def read(self, nrows=None):
   2036         try:
-> 2037             data = self._reader.read(nrows)
   2038         except StopIteration:
   2039             if self._first_chunk:

pandas\_libs\parsers.pyx in pandas._libs.parsers.TextReader.read()

pandas\_libs\parsers.pyx in pandas._libs.parsers.TextReader._read_low_memory()

pandas\_libs\parsers.pyx in pandas._libs.parsers.TextReader._read_rows()

pandas\_libs\parsers.pyx in pandas._libs.parsers.TextReader._tokenize_rows()

pandas\_libs\parsers.pyx in pandas._libs.parsers.raise_parser_error()

ParserError: Error tokenizing data. C error: Expected 1 fields in line 4, saw 2
```

**Comment by blahem** [comment]
Can you add a 1-sec cooldown (time.sleep(1)) after each request and tell me what happens?

**Comment by r8drascal** [comment]
@blahem Thanks that actually worked! I'm also

**Comment by blahem** [comment]
@r8drascal Happy to help! I had the same problem, and just added a 1 second delay after each request. And the second thing I did is to wait 15 seconds as soon as a parser error appears. With this I don't have any problem.
I'm not sure, but I think in the past the yahoo api took some time to respond to requests, like to avoid spam. Now it doesn't, and this allows you to send 7-8 requests within a second, and when you do that they block your requests...
I say this because I noticed that the data was downloaded much faster than before.

blahem
